Compiler/FrontEnd/Inline.mo

```diff
@@ -1110,71 +1110,71 @@ end addTplAssignToRepl;
-      then
-        (oInputs,oOutput,oBody,repl);
+  for elt in fn loop
+   _ := match(elt)
 
-    case (DAE.VAR(componentRef=cr,direction=DAE.OUTPUT(), binding=binding as SOME(exp))::rest,_,_,_,_)
+     case DAE.VAR(componentRef=cr,direction=DAE.INPUT())
+         equation
+           oInputs = cr::oInputs;
+         then ();
+
+    case DAE.VAR(componentRef=cr,direction=DAE.OUTPUT(), binding=binding as SOME(exp))
       equation
         // use type of cref, since var type is different
         // and has no hint on array or record type
         tp = ComponentReference.crefTypeFull(cr);
-        st = listAppend(iBody, {DAE.STMT_ASSIGN(exp1 = Expression.crefExp(cr), exp = exp, source=DAE.emptyElementSource, type_=tp)});
-        (oInputs,oOutput,oBody,repl) = getFunctionInputsOutputBody(rest,iInputs,cr::iOutput,st,iRepl);
-      then
-        (oInputs,oOutput,oBody,repl);
-
+        oBody = listAppend(oBody, {DAE.STMT_ASSIGN(exp1 = Expression.crefExp(cr), exp = exp, source=DAE.emptyElementSource, type_=tp)});
+        oOutputs = cr :: oOutputs;
+       then ();
 
-    case (DAE.VAR(componentRef=cr,direction=DAE.OUTPUT(), binding=NONE())::rest,_,_,_,_)
+    case DAE.VAR(componentRef=cr,direction=DAE.OUTPUT(), binding=NONE())
       equation
-        (oInputs,oOutput,oBody,repl) = getFunctionInputsOutputBody(rest,iInputs,cr::iOutput,iBody,iRepl);
-      then
-        (oInputs,oOutput,oBody,repl);
+        oOutputs = cr :: oOutputs;
+       then ();
 
-    case (DAE.VAR(componentRef=cr,protection=DAE.PROTECTED(),binding=binding)::rest,_,_,_,_)
+    case DAE.VAR(componentRef=cr,protection=DAE.PROTECTED(),binding=binding)
       equation
         // use type of cref, since var type is different
         // and has no hint on array or record type
         tp = ComponentReference.crefTypeFull(cr);
         false = Expression.isArrayType(tp);
         false = Expression.isRecordType(tp);
-        repl = addOptBindingReplacements(cr,binding,iRepl);
-        (oInputs,oOutput,oBody,repl) = getFunctionInputsOutputBody(rest,iInputs,iOutput,iBody,repl);
+        oRepl = addOptBindingReplacements(cr,binding,oRepl);
       then
-        (oInputs,oOutput,oBody,repl);
-    case (DAE.ALGORITHM(algorithm_ = DAE.ALGORITHM_STMTS(st))::rest,_,_,_,_)
+        ();
+
+    case DAE.ALGORITHM(algorithm_ = DAE.ALGORITHM_STMTS(st))
       equation
-        st = listAppend(iBody,st);
-        (oInputs,oOutput,oBody,repl) = getFunctionInputsOutputBody(rest,iInputs,iOutput,st,iRepl);
+        oBody = listAppend(oBody,st);
       then
-        (oInputs,oOutput,oBody,repl);
-    case (elt::rest,_,_,_,_)
+        ();
+
+    case _
       algorithm
         Error.addInternalError("Unknown element: " + DAEDump.dumpElementsStr({elt}), sourceInfo());
       then fail();
-  end match;
+    end match;
+
+
+  end for;
+
+  oInputs := listReverse(oInputs);
+  oOutputs := listReverse(oOutputs);
+
```

Comment:
Is it necessary to reverse the list? you used listAppend() before, so I guess the order is not the same as the old version anyway...

Reply:
The order should be importend, because
` newExp = Expression.makeTuple(list( VarTransform.getReplacement(repl,cr) for cr in lst_cr)); // lst_cr = oOutputs
`
will transform
`(a, b) = f(c) -> a=f(c)[2], b = f(c)[1]
`
, but maybe I wrong..